patches/tModLoader/Terraria/Main.cs.patch

```diff
@@ -3671,7 +3671,7 @@
 -						num5 *= (float)dayRate;
 -						if (rand.Next((int)num5) == 0)
 +						num5 *= (float)desiredWorldEventsUpdateRate;
-+						if (rand.NextFloat() <= 1f / num5)
++						if (rand.NextFloat() <= 1f / Math.Max(num5, 2400))
```

Comment:
I checked the surrounding code. `UpdateWeather` is called once per day 'tick' and if the day is running super slow, your other changes will ensure the calculations are correct. You should actually do `num5 *= Math.Max(desiredWorldEventsUpdateRate, 1)`